### ISSUE TITLE
Models are not updated

### ISSUE THREAD

**Issue by mathiasewald** [fragaria/angular-daterangepicker#42]
Hi, please look at Plunker http://plnkr.co/edit/uZwZl1XOdWYHiCji7YPQ The date range picker is not updating the model. Therefore clicking + - Button results in complete reset of the values.

**Comment by ReinierGielen** [comment]
I have a different situation, but also a problem with the model not being updated consistently.

On a new screen if the first thing you do is change the date with this widget the model IS updated.

If I do something to another field first it is NOT updated.

very strange and a big problem.

**Comment by ReinierGielen** [comment]
UPDATE: seem that in an intial basic configuration it works fine. I get problems when I update the options object and reset the options. When I redefine the entire options object it seems to work but when I only update the ranges the model is not updated.

**Comment by bborowin** [comment]
I'm having what appears to be a related problem - here's a bit of code to illustrate the issue:

          <input
            date-range-picker
            readonly="readonly"
            class="col-md-3 form-control date-picker"
            type="text"
            ng-model="date"
            options="dateRangePickerOptions" />
          <button
            class="btn btn-primary"
            ng-click="date = null">
            Clear
          </button>

The clear button works (clears the `date` model and thus the text box), but when I open the dropdown, the old range is still highlighted. After selecting the same range again, clicking the Clear button does not have any effect (presumably because `date === null` already), and the range dates remain in the text box.

**Comment by marcelinhov2** [comment]
I had the same issue as you, but I did so many things that I don't remember. But I will try to help you anyway.

First thing, I have created a method called datepicker. I use it just to set the default values of the daterangepicker.

datepicker: (from, to) =>
    @$scope.datepicker = {}
    @$scope.datepicker.ranges = []
    @$scope.datepicker.range = 
      startDate: if from and from.isValid() then from else null
      endDate: if to and to.isValid() then to else null

When I need to clean the date ( for example, a button to empty everything ) I just call this method and the magic happens.

Besides, I have a watcher too.

@$scope.$watch "datepicker.range", ((now, was, scope) =>
      if now.startDate and now.endDate
        scope.form.valid_from = moment(now.startDate).format('YYYY-MM-DD')
        scope.form.valid_to = moment(now.endDate).format('YYYY-MM-DD')
      else
        scope.form.valid_from = null
        scope.form.valid_to = null
), true

I just replicate the values from @$scope.datepicker to the @$scope.form using the date format that I need to send to my webservice.

Summarizing, you will need to have 2 different objects, one for the daterangepicker and the other for your controller. If you use the same object, you will have the problem that you describe.

P.S.: I'm sorry for the explanation but I have a lot of difficult to teach. It makes more sense in my head. I glad it will help you.
